toolkit/tools/liveinstaller/liveinstaller.go

```diff
@@ -119,19 +125,69 @@ func installerFactory(forceAttended bool, configFile, templateConfigFile string)
 	}
 
 	if isAttended {
-		installFunc = func(args imagerArguments) (bool, error) {
+		installFunc = func(args imagerArguments) (installationDetails, error) {
 			return terminalUIAttendedInstall(templateConfigFile, args)
 		}
 	} else {
-		installFunc = func(args imagerArguments) (bool, error) {
+		installFunc = func(args imagerArguments) (installationDetails, error) {
 			return unattendedInstall(configFile, args)
 		}
 	}
 
 	return
 }
 
-func updateBootOrder() (err error) {
+func updateBootOrder(installDetails installationDetails) (err error) {
+	if (installDetails.finalConfig.DefaultSystemConfig.BootType != "efi") {
+		logger.Log.Info("No BootType of 'efi' detected. Not attempting to set EFI boot order.")
+		return
+	}
+
+	err = removeOldMarinerBootTargets()
+	if (err != nil) {
+		return
+	}
+
+	logger.Log.Info("Setting Boot Order")
+	const squashErrors = false
+	program, commandArgs := formatBootEntryCreationCommand(installDetails)
+	err = shell.ExecuteLive(squashErrors, program, commandArgs...)
+
+	return
+}
+
+func formatBootEntryCreationCommand(installDetails installationDetails) (program string, commandArgs []string) {
+	program = "efibootmgr"
```

Comment:
This logic seems sound to me. Question: how do we know that the userspace app efibootmgr will be installed at the point that the installer is running? I think I'm misunderstanding when this liveinstaller.go code is running.

Reply:
runliveinstaller runs liveinstaller.go and is an initrd environment created with the packages from iso-initrd-packages.json.   It seems efibootmgr has been part of that package list for a while, which thankfully meant we don't need to add it :) thank you for checking!

Reply:
Ah, that clears things up. Thanks for the info!